### PR TITLE
fix: restrict the range of index when moving through variations

### PR DESCRIPTION
**Pull request by xiaoyifang**
When move through the node variations ,  restrict the index goes too far from the end.

### CODE REVIEW

File: katrain/gui/badukpan.py [hunk 1]
```diff
@@ -1053,7 +1053,7 @@ def draw_hover_contents(self, *_args):
                 animating_pv = self.animating_pv
                 if animating_pv:
                     pv, node, start_time, _ = animating_pv
-                    up_to_move = self.get_animate_pv_index()
+                    up_to_move = min( len(pv), self.get_animate_pv_index())
```
Comment (sanderland) [inline katrain/gui/badukpan.py:1056]: this seems redundant

Comment (xiaoyifang) [reply katrain/gui/badukpan.py:1056]: This is necessary.
Withouth this, the `self.get_animate_pv_index` may return a very large number ,eg 100 (depends on how much time has been elapsed).

When using mouse scroll to move back the variations ,you have to scroll 100 times to reach to the beginning. (though most of them has been triggered by break and return immediately).

Comment (xiaoyifang) [reply katrain/gui/badukpan.py:1056]: the two lines changes can be kept only one of them should also be ok.

Comment (sanderland) [reply katrain/gui/badukpan.py]: I thought just a change to get_animate_pv_index would be enough - did you remove the wrong one? 🤔

Comment (xiaoyifang) [reply katrain/gui/badukpan.py]: The current changed line is in the `get_animate_pv_index ` already.

**Pull request merged**